ref/rust/src/lib.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2017 Clark Moody
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#![warn(missing_docs)]
+
+//! Encoding and decoding Bech32 Bitcoin Segwit Addresses
+//! 
+//! Encoding and decoding for Bech32 strings and Bitcoin Segregated Witness 
+//! addresses. Bech32 is a 5-bit (base-32) encoding scheme that produces
+//! strings that comprise a human-readable part, a separator, a data part,
+//! and a checksum. The encoding implements a BCH code that guarantees
+//! error detection of up to four characters with less than 1 in 1 billion
+//! chance of failing to detect more errors.
+//! 
+//! The library contains `bech32` utilities for generic encoding of Bech32
+//! strings and `wit_prog` for converting witness programs to Bitcoin
+//! addresses and back.
+//! 
+//! # Examples
+//! 
+//! ```rust
+//! use bech32::wit_prog::WitnessProgram;
+//! 
+//! let witness_program = WitnessProgram {
+//!     version: 0,
+//!     program: vec![
+//!                 0x00, 0x00, 0x00, 0xc4, 0xa5, 0xca, 0xd4, 0x62, 
+//!                 0x21, 0xb2, 0xa1, 0x87, 0x90, 0x5e, 0x52, 0x66, 
+//!                 0x36, 0x2b, 0x99, 0xd5, 0xe9, 0x1c, 0x6c, 0xe2, 
+//!                 0x4d, 0x16, 0x5d, 0xab, 0x93, 0xe8, 0x64, 0x33]
+//! };
+//! 
+//! let enc_result = witness_program.to_address("tb".to_string());
+//! assert_eq!(enc_result.unwrap(), 
+//!     "tb1qqqqqp399et2xygdj5xreqhjjvcmzhxw4aywxecjdzew6hylgvsesrxh6hy".to_string());
+//! ```
+
+pub mod bech32;
+pub mod wit_prog;
+
+/// Error types for Bech32 encoding / decoding
```

Comment:
I would like to publish as a crate later. My fork may diverge at that point to add the extras to make it happen.